scripts/management/xcash-maintenance

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+
+# Adjust this variable to the name of your xcash-daemon unit file name
+XCASH_DAEMON_SRV=xcash-daemon
+
+# Adjust this variable to the name of your xcash-wallets unit file name
+XCASH_WALLET_SRV=xcash-rpc-wallet
+
+# Adjust this variable to the name of your xcash-dpops unit file name
+XCASH_DPOPS_SRV=xcash-dpops
+
+# Adjust this variable to the name of your mongodb unit file name
+MONGODB_SRV=mongodb
+
+XCASH_DAEMON_LOG_FILE=/var/x-network/logs/blockchain.log
+XCASH_WALLET_LOG_FILE=/var/x-network/logs/rpc.log
+### CUSTOM VARIABLES ###
+
+DAEMON_URL="http://127.0.0.1:18281/json_rpc"
+WALLET_URL="http://127.0.0.1:18285/json_rpc"
+DPOPS_QUERY="curl -s -H \"Content-Type: application/json\" -X GET http://127.0.0.1:18283"
+CURL='curl -w "\n" -s -H "Content-Type: application/json" -X'
+
+red=`tput setaf 1`
+green=`tput setaf 2`
+reset=`tput sgr0`
+
+PS3='
+Please enter an option: '
+
+options=(
+	"Check blockchain height"
+	"Get node statistics"
+	"Get node balance"
+	"Check services status"
+	"Start stopped services"
+    "Stop all services"
+	"Restart all services"
+	"Get X-Cash DPoPS log"
+    "Get X-Cash Daemon log"
+    "Get X-Cash RPC Wallet log"
+	"Quit"
+)
+
+function check_height(){
+	if lsof -Pi :18281 -sTCP:LISTEN -t >/dev/null ; then
+        DATA=`$CURL GET $DAEMON_URL -d '{"jsonrpc":"2.0","id":"0","method":"get_block_count"}'`
+
+		HEIGHT=$(echo "$DATA" | grep '"count"' | awk {'print $2'} | sed s'|,||g' | tr -d $'\r')
+		STATUS=$(echo "$DATA" | grep '"status"' | awk {'print $2'} | sed s'|"||g' | tr -d $'\r')
+		
+		if [ $STATUS == "OK" ]; then
+			STATUS=${green}OK${reset}
+		else 
+			SATTUS=${red}FAIL${reset}
+		fi
+
+		echo
+		echo "Blockhain sync status: $STATUS"
+		echo "Blockchain height: $HEIGHT"
+	else
+		echo
+		echo "X-Cash Daemon service is not running...."
+		echo "Check the $XCASH_DAEMON_SRV service status"
+	fi
+}
```

Comment:
This returns the status field of the "get_block_count" RPC call.

The status field for that RPC call, always returns true if the function works. The status field also does not indicate the actual blockchain synchronization status.

Ideally the output to the RPC call get_info reading the target and target_height fields and comparing them with the delegates block_height can tell you the blockchain status, but these are buggy where if synced they can return 0, and sometimes will return false positives for not synced.

The best way I have found to give an accurate true or false if they are fully synced is to use a method I use in the DPOPS code for [check_if_blockchain_is_fully_synced](https://github.com/X-CASH-official/xcash-dpops/blob/master/src/functions/network_functions/network_daemon_functions/network_daemon_functions.c#L32)

You just need to
#1 get the current block height for the delegate (using curl and the get_block_count)
#2 get the current block height of the 5 seed nodes (using curl and the get_block_count)
#3 check if the current height is less than the seed node height for each seed node
#4 If the current height is less than 3/5 of the seed nodes, then they are not fully synced

Reply:
I will apply these changes, tbh, was just a "is the service listening on the port" check.